test/session_tests/DocumentQueryTest.ts

```diff
@@ -8,220 +8,493 @@ import {RequestExecutor} from "../../src/Http/Request/RequestExecutor";
-      session = store.openSession();
+    it('should query with whereLessThan with :p', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true,
+          queryParameters: {
+            "p0" : 6
+          }
+        })
+        .whereLessThan<string>('order', ':p0').get();
+
+      expect(results).to.have.lengthOf(5);
 
-      await session.query({
-        waitForNonStaleResults: true, 
-        includes: ['product_id']
-      }).where({uid: 92}).get();
+      results.map(function (obj) {
+        expect(obj.order).to.not.be.greaterThan(6);
+        expect(obj.order).to.be.lessThan(6);
+      });
 
-      await session.load('Product/108');
-      expect(session.numberOfRequestsInSession).to.equals(1);            
     });
 
-    it('should query with nested objects', async() => {
-      const results: Company[] = await store.openSession().query<Company>({
-        documentType: Company,
-        nestedObjectTypes: {product: Product}
-      }).whereEquals<string>('name', 'withNesting').get();
+    it('should query by startsWith', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .startsWith<string>('name', 'wi').get();
+
+      expect(results).to.have.lengthOf(10);
+
+      results.map(function (obj) {
+        expect(obj).to.include({'name': 'withNesting'});
+        expect(obj.name).to.contain('wi');
+      });
 
-      expect(results[0].product).to.be.instanceOf(Product);
-      expect(results[0]).to.be.instanceOf(Company)
     });
 
-    it('should make query with fetch terms', async() => {
-      const results: Product[] = await store.openSession().query<Product>({
-        documentType: Product,
-        waitForNonStaleResults: true,
-        indexName: 'Testing_Sort'
-      })
-      .whereBetweenOrEqual<number>('uid', 2, 4).selectFields('doc_id').get();
-      
-      expect(_.every(results, (result: Product) => result.hasOwnProperty('doc_id'))).to.be.true;
+    it('should query by endsWith', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .endsWith<string>('name', 'ing').get();
+
+      expect(results).to.have.lengthOf(10);
+
+      results.map(function (obj) {
+        expect(obj).to.include({'name': 'withNesting'});
+        expect(obj.name).to.contain('ing');
+      });
+
     });
-  });  */
+
+    it('should query by search', async () => {
+      const results: Universal[] = await store.openSession()
+        .query<Universal>({
+          indexName: 'Universals',
+          WaitForNonStaleResults: true
+        })
+        .search('name', 'withNesting', 'order', 2, '=', 2).get();
```

Comment:
This call is hardly understandable. What's happening here?

Reply:
I believe it's made this way due to lack of tokenization and builder.. This however makes API inconsistent. We need it like this:
https://github.com/ravendb/ravendb/blob/v4.0/src/Raven.Client/Documents/Session/IDocumentQueryBase.cs#L349
```
 public search<V extends RQLValue>(field, value, boostField, boostValue, count) {

    this._builder
      .from(this.indexName, this.fromCollection)
      .where('search', field, value)
      .or
      .where('boost', {boostField: boostField, boostValue: boostValue}, count);

    return this;
  }
```